File: src/main/kotlin/user/ancyle/chrms/entities/concretes/JobSeeker.java
package user.ancyle.chrms.entities.concretes;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name="job_seekers")
@Data
public class JobSeeker {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="js_id")
    private short jsId;

    @Column(name="first_name")
    private String firstName;

    @Column(name="last_name")
    private String lastName;

    @Column(name="national_id")
    private String nationalId;

    @Column(name="birth_year")
    private short birthYear;

    @OneToOne(cascade = CascadeType.ALL)
    private User user;
}

File: src/main/kotlin/user/ancyle/chrms/entities/concretes/JobSeeker.kt
package user.ancyle.chrms.entities.concretes

import java.util.*
import javax.persistence.*


@Entity
@Table(name = "job_seekers")
public data class JobSeeker(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "js_id")
    val jsId: Int,
    @Column(name = "js_no")
    val jsNo: String,
    @Column(name = "js_mail")
    val jsMail: String,
    @Column(name = "js_pass")
    val jsPass: String,
    @Column(name = "js_first_name")
    val jsFirstName: String,
    @Column(name = "js_last_name")
    val jsLastName: String,
    @Column(name = "js_national_identity")
    val nationalIdentity: String,
    @Column(name = "js_birth_year")
    val jsBirthYear: Int,
    @Column(name = "is_active")
    val isActive: Boolean,
    @Column(name = "created_date")
    val createdDate: Date
)
